api/src/main/java/com/vimeo/networking2/utils/ApiResponseErrorConverter.kt

```diff
@@ -0,0 +1,26 @@
+package com.vimeo.networking2.utils
+
+import com.vimeo.networking2.ApiError
+import retrofit2.Response
+import retrofit2.Retrofit
+
+/**
+ * Converter for parsing an error from the API into a [ApiError] object.
+ *
+ * @param retrofit Required for using the response body converter to do the conversion.
+ */
+class ApiResponseErrorConverter(private val retrofit: Retrofit) {
```

Comment:
I'd suggest you extract an interface for this class and change `ApiRequestRetrofitCallback` and `Authenticator` to take the interface. That allows those classes to be more easily tested.